serde/src/de/mod.rs

```diff
@@ -641,10 +785,26 @@ pub trait SeqVisitor {
     /// The error type that can be returned if some error occurs during deserialization.
     type Error: Error;
 
-    /// This returns a `Ok(Some(value))` for the next value in the sequence, or `Ok(None)` if there
-    /// are no more remaining items.
+    /// This returns `Ok(Some(value))` for the next value in the sequence, or
+    /// `Ok(None)` if there are no more remaining items.
+    ///
+    /// `Deserialize` implementations should typically use `SeqVisitor::visit`
+    /// instead.
+    fn visit_seed<T>(&mut self, seed: T) -> Result<Option<T::Value>, Self::Error>
+        where T: DeserializeSeed;
+
+    /// This returns `Ok(Some(value))` for the next value in the sequence, or
+    /// `Ok(None)` if there are no more remaining items.
+    ///
+    /// This method exists as a convenience for `Deserialize` implementations.
+    /// `SeqVisitor` implementations should not need to override the default
```

Comment:
I can't think of any reason.

The three places to put this function are:

- Free function like serde::ser::visit_seq_element(seq_visitor),
- Separate trait where we impl<T> SeqVisitorExt for T where T: SeqVisitor,
- Here.

I think of the three, this is the best tradeoff.

- I don't like the first option because if you write a method that takes in a SeqVisitor, your preference is going to be to call methods on the SeqVisitor. I would prefer for Deserialize implementations to deal with the non-seed method wherever possible and putting them here at least puts them on equal footing with the seed methods.
- I don't like the second option because Deserialize will need to `use` an extra trait before they can use the methods and again they will prefer not doing that.

Deserializer implementors will be much more likely to be reading our rustdoc than Deserialize implementors, so a doc comment here that says you don't need to override the behavior counteracts most of the disadvantage.